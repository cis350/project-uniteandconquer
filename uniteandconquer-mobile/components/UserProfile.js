import React from 'react';
import {
  StyleSheet, View, Button, Text, ScrollView, TouchableOpacity,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import Notification from './Notification';

// styling ---------

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FAE9C7',
    height: '100%',
    width: '100%',
  },
});

const userStyles = StyleSheet.create({
  container: {
    alignItems: 'stretch',
    justifyContent: 'center',
    margin: 10,
  },
  text: {
    textAlign: 'right',
    margin: 4,
    fontSize: 18,
    marginBottom: 10,
  },
  button: {
    alignSelf: 'flex-end',
  },
});

const profileContainer = StyleSheet.create({
  header: {
    fontSize: 35,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  buttons: {
    display: 'flex',
    flexDirection: 'row',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
  },
  leftButton: {
    backgroundColor: '#FFD9A0',
    height: 35,
    width: 150,
    marginRight: 35,
  },
  rightButton: {
    backgroundColor: '#FFD9A0',
    height: 35,
    width: 150,
    marginLeft: 35,
  },
  subtitle: {
    marginTop: 5,
    fontSize: 25,
    textAlign: 'center',
    fontStyle: 'italic',
  },
  post: {
    marginTop: 10,
    padding: 10,
    backgroundColor: '#FFD9A0',
    marginBottom: 5,
    width: 380,
  },
  postHeader: {
    fontSize: 19,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  postContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  showButton: {
    position: 'absolute',
    top: 0,
    left: 0,
    margin: 10,
  },
});

const tagStyles = StyleSheet.create({
  container: {
    padding: 10,
    backgroundColor: '#FFD9A0',
    marginBottom: 15,
    width: 380,
    marginTop: 15,
  },
  header: {
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: 19,
    fontStyle: 'italic',
    marginBottom: 10,
    textAlign: 'center',
  },
  tags: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  tag: {
    margin: 10,
    backgroundColor: '#FFB443',
    borderRadius: 15,
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderWidth: 1,
    overflow: 'hidden',
  },
  position: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

// app content --------

export default function UserProfile({ navigation }) {
  const [showNotif, setShowNotif] = React.useState(false);
  return (
    <ScrollView style={styles.container}>
      {showNotif && <Notification setShowNotif={setShowNotif} showNotif={showNotif} />}
      <View style={userStyles.container}>
        <TouchableOpacity
          style={{ alignSelf: 'flex-end' }}
          onPress={() => navigation.navigate('UserProfile')}
        >
          <Icon name="user" size={28} style={userStyles.icon} />
        </TouchableOpacity>
        <Text style={userStyles.text}>
          Hello, guest!
        </Text>
        <Text style={userStyles.button}>
          <Button
            color="#000"
            title="My Chats"
          />
        </Text>
        <Text style={userStyles.button}>
          <Button
            color="#000"
            title="Log In"
            onPress={() => navigation.navigate('LogIn')}
          />
        </Text>
      </View>
      <View style={profileContainer.showButton}>
        <TouchableOpacity
          onPress={() => setShowNotif(!showNotif)}
        >
          <Icon
            name="bell"
            size={25}
          />
        </TouchableOpacity>
      </View>
      <View>
        <Text style={profileContainer.header}>My Profile</Text>
        <View style={profileContainer.buttons}>
          <View style={profileContainer.leftButton}><Button color="#000" title="Settings" /></View>
          <View style={profileContainer.rightButton}><Button color="#000" title="Main Page" onPress={() => navigation.navigate('Home')} /></View>
        </View>
        <Text style={profileContainer.subtitle}>My Owned Posts</Text>
        <View style={profileContainer.postContainer}>
          <View style={profileContainer.post}>
            <Text style={profileContainer.postHeader}>This is a Post</Text>
            <Text style={profileContainer.postContent}>
              This post is led by Jeremy and trades [ITEM] for $[VALUE] with
              maturity [DATE]
            </Text>
          </View>
          <View style={profileContainer.post}>
            <Text style={profileContainer.postHeader}>This is a Post</Text>
            <Text style={profileContainer.postContent}>
              This post is led by Jeremy and trades [ITEM] for $[VALUE] with
              maturity [DATE]
            </Text>
          </View>
          <View style={profileContainer.post}>
            <Text style={profileContainer.postHeader}>This is a Post</Text>
            <Text style={profileContainer.postContent}>
              This post is led by Jeremy and trades [ITEM] for $[VALUE] with
              maturity [DATE]
            </Text>
          </View>
        </View>
        <Text style={profileContainer.subtitle}>My Joined Posts</Text>
        <View style={profileContainer.postContainer}>
          <View style={profileContainer.post}>
            <Text style={profileContainer.postHeader}>This is a Post</Text>
            <Text style={profileContainer.postContent}>
              This post is led by Jeremy and trades [ITEM] for $[VALUE] with
              maturity [DATE]
            </Text>
          </View>
          <View style={profileContainer.post}>
            <Text style={profileContainer.postHeader}>This is a Post</Text>
            <Text style={profileContainer.postContent}>
              This post is led by Jeremy and trades [ITEM] for $[VALUE] with
              maturity [DATE]
            </Text>
          </View>
          <View style={profileContainer.post}>
            <Text style={profileContainer.postHeader}>This is a Post</Text>
            <Text style={profileContainer.postContent}>
              This post is led by Jeremy and trades [ITEM] for $[VALUE] with
              maturity [DATE]
            </Text>
          </View>
        </View>
        <View style={tagStyles.position}>
          <View style={tagStyles.container}>
            <Text style={tagStyles.header}>My Interests</Text>
            <View style={tagStyles.tags}>
              <Text style={tagStyles.tag}>tag</Text>
              <Text style={tagStyles.tag}>tag</Text>
            </View>
          </View>
        </View>
      </View>
    </ScrollView>
  );
}
