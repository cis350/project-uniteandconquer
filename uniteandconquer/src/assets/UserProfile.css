
td,
th {
  border: 10px solid #ffd9a0;
  box-sizing: border-box;
  max-width: 430px;
  width: 430px;
  min-width: 430px;
}

table {
  border-collapse: collapse;
  margin-left: 50px;
}

th {
  font-size: 25px;
}

td {
  padding: 15px;
}

td:hover {
  background-color: #ffd9a0;
}

.post-title {
  font-size: 24px;
  margin-bottom: 20px;
}

.post-content {
  font-size: 17px;
  max-width: 400px;
  width: 400px;
  min-width: 400px;
}

.user-profile-page {
  display: grid;
  grid-template-columns: 0.2fr 0.9fr;
}

.profile-title {
  font-size: 25px;
  text-align: center;
}

.notifications-pos {
  position: absolute;
  top: 0;
  right: 0;
}

.bell-pos {
  position: absolute;
  top: 0;
  right: 0;
  padding: 50px;
}

.bell-button {
  background: none;
  border: none;
}

.bell-button:hover {
  color: rgb(133, 235, 85);
}