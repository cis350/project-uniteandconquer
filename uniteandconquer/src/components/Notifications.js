import React from 'react';
import '../assets/Notifications.css';

const notifyDB = require('../modules/NotificationDB');

const myStorage = window.sessionStorage;
const userID = myStorage.getItem('UserID');

function Notifications({
  showNotifs, setShowNotifs, notifs, setNotifs,
}) {
  const messageGenerator = () => notifs.map(
    (notif) => {
      const message = notif.content;
      const date = new Date(notif.createdAt).toLocaleString('en-US', { timeZone: 'America/New_York' });
      return (
        <div>
          {message}
          {' '}
          {date}
        </div>
      );
    },
  );
  const handleClick = () => {
    notifyDB.deleteNotifications(userID, notifs, (success, err) => {
      if (success) {
        console.log('click');
        setNotifs([]);
      } else {
        console.log(err);
      }
    });
    setShowNotifs(!showNotifs);
  };

  return (
    <div className="notifications-container">
      <div className="toggle-notifs">
        <button className="notif-button" type="button" onClick={handleClick}>
          {' '}
          <i className="fas fa-times fa-2x" />
        </button>

      </div>
      <div className="notifications-title">Notifications</div>

      {messageGenerator()}

    </div>

  );
}

export default Notifications;
