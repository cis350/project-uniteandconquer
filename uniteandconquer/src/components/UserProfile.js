import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import SidebarTags from './SidebarTags';
import Notifications from './Notifications';
import '../assets/UserProfile.css';

const UserDB = require('../modules/UserDB');

function UserProfile() {
  const [showNotifs, setShowNotifs] = useState(false);
  /** used for showing interest in side bar */
  const [tags, setTags] = useState([]);
  /** list of posts and whish list that will show on profile page */
  const [posts, setPosts] = useState([]);
  // eslint-disable-next-line no-unused-vars
  const [wishList, setWishList] = useState([]);

  const myStorage = window.sessionStorage;
  const PostTableGenerator = (postsOrWishList) => {
    const table = postsOrWishList.map((post) => {
      const url = `/post-details/${post.id}`;
      return (
        <div>
          <tr className="table-post">
            <td className="table-post-content">
              <Link className="link" to={url}>
                <div className="post-title">
                  Post:
                  {' '}
                  <br />
                  {' '}

                  {post.itemName}
                  - [
                  {post.status}
                  ]
                </div>
                <div className="post-content">
                  This post is led by [
                  {post.ownerName}
                  ] and trades [
                  {post.itemNumTarget}
                  ] of [
                  {post.itemNumCurrent}
                  ] for
                  $[
                  {post.pricePerItem}
                  ]
                  <br />
                  <br />
                  [
                  {post.tags}
                  ]
                </div>
              </Link>
            </td>
          </tr>
        </div>
      );
    });
    return table;
  };

  useEffect(() => {
    const userID = myStorage.getItem('UserID');
    UserDB.getUserDetails(userID, (success, userInfo, err) => {
      if (success) {
        setTags(userInfo.interests);
        setPosts(userInfo.posts);
        setWishList(userInfo.wishList);
      } else {
        console.log(err);
      }
    });
  }, []);

  return (
    <div className="user-profile-page">
      <SidebarTags tags={tags} />
      <div>
        <div className="profile-title"><h1>My Profile</h1></div>
        <div className="bell-pos">
          <button className="bell-button" type="button" onClick={() => setShowNotifs(!showNotifs)}>
            {' '}
            <i className="fas fa-bell fa-2x" />
          </button>
        </div>
        <div className="notifications-pos">{showNotifs ? <Notifications showNotifs={showNotifs} setShowNotifs={setShowNotifs} /> : ''}</div>
        <div className="table-lists">
          <table>
            <thead>
              <tr>
                <td>My Active Posts</td>
              </tr>
            </thead>
            <tbody>
              { PostTableGenerator(posts)}
            </tbody>
          </table>
          <table>
            <thead>
              <tr>
                <td>My Wishlist</td>
              </tr>
            </thead>
            <tbody>
              { PostTableGenerator(wishList)}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}

export default UserProfile;
