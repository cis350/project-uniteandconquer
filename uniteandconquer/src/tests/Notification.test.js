/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import Notifications from '../components/Notifications';
import UserProfile from '../components/UserProfile';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('open and hide notifications', () => {
  render(<UserProfile />);
  fireEvent.click(screen.getByTestId('show-notifications'));
  fireEvent.click(screen.getByTestId('hide-notifications'));
});

// test('Notifications title', () => {
//   render(<Notifications />);
//   const title = screen.getByText('Notifications');
//   expect(title).toBe('Notifications');
// });

test('snapshot notifications', () => {
  const component = renderer.create(<Notifications />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
