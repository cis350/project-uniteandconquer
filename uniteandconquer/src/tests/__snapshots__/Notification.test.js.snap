// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot profile 1`] = `
<div
  className="user-profile-page"
>
  <div
    className="sidebar"
  >
    <div
      className="greeting-box"
    >
      <i
        className="far fa-user-circle fa-2x"
      />
       
      Hi,
      Guest
    </div>
    <div
      className="home-box"
    />
    <div
      className="settings-box"
    />
    <div
      className="preferences-box"
    >
      <div
        className="preferences-title"
      >
        My Preferences
      </div>
    </div>
  </div>
  <div>
    <div
      className="profile-title"
    >
      <h1>
        My Profile
      </h1>
    </div>
    <div
      className="bell-pos"
    >
      <button
        className="bell-button"
        data-testid="show-notifications"
        onClick={[Function]}
        type="button"
      >
         
        <i
          className="fas fa-bell fa-2x"
        />
      </button>
    </div>
    <div
      className="notifications-pos"
    >
      
    </div>
    <div
      className="table-lists"
    >
      <table>
        <thead>
          <tr>
            <td>
              My Active Posts
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
      <table>
        <thead>
          <tr>
            <td>
              My Wishlist
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
